spring-xd-messagebus-rabbit/src/main/java/org/springframework/xd/dirt/integration/rabbit/RabbitMessageBus.java

```diff
@@ -874,4 +878,30 @@ public boolean getAutoBindDLQ(boolean defaultValue) {
 
 	}
 
+	@Override
+	public void storeForManualAck(MessageHeaders headers) {
+		if (headers.containsKey("amqp_channel")) {
+			Channel channel = (com.rabbitmq.client.Channel) headers.get("amqp_channel");
+			if (!channelsForManualAck.containsKey(channel)) {
+				channelsForManualAck.put(channel, 1);
+			}
+			else {
+				channelsForManualAck.put(channel, channelsForManualAck.get(channel) + 1);
```

Comment:
You can't use a simple incrementing value; you need to retrieve the delivery tag from the headers (`amqp_deliveryTag`). Also, instead of hard-coded strings, you should use the `AmqpHeaders` constants.